### PR TITLE
Add scrambler for load testing

### LINKED ISSUE #376: Refactor `expand_test_data.py` to accept PIIRecords
## Summary
Load testing data will need to be scrambled to account for some matches and some non-matches. The `expand_test_data.py` script does a nice job of scrambling data but it only accepts CSV data that comes from NBS while the load testing data will be `PIIRecods`. We need to update the script to abstract away the CSV-specific parts so that the `apply_field_scrambling` function can accept other formats.

## Acceptance Criteria
- [ ] `expand_test_data.py` is refactored to work on a variety on input data (not just the NBS CSV data)
- [ ]  For the purposes of load testing, it can accept an individual PIIRecord, multiply it approximately 5 times (we need roughly a 5:1 ratio of records to individuals to replicate Montana's data), scramble some of the data, and return it

## Details / Tasks
Note: for the purposes of load testing, it is okay if the individual records are scrambled such that they no longer match the original record.

### PR DESCRIPTION
**Pull request by m-goggins**
## Description
This PR adds a script to scramble generated records for the purposes of load testing. It re-uses much of the code @bamader wrote for the scrambling the csv of NBS data, and adapts it for the json/PIIRecord data we have. To test out the code, you can  run `python3 -m tests.load.scripts.scramble_data --file="./tests/load/assets/test_data.json"` and it will spit out a file with scrambled records and the original from the `test_data.json` file. 

## Related Issues
Closes #376 

## Additional Notes
If there is time after load testing to refactor this, there's a good amount of cleanup/refactoring I'd like to do to make this more adaptable to other file types, e.g., the NBS CSV and other CSVs. But, given the priority for load testing (not scrambling), this script does the job and does it quickly.

Background for why we need the script:
We want to be able to seed the test MPI during load testing with a variety of records that may or may not match each other. We plan to:
1. generate 1.5 million records (the "original" data)
2. grab ~1 million of the records from the original data and scramble/multiple them using this script to get roughly 4-5 million records (the "scrambled" data)
3. Seed the MPI with the scrambled data
4. Randomly select records from the original data to hit the `/link` endpoint with to see how the database performs with a larger MPI

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: tests/load/scrambler/utils.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import string
+import typing
+
+from . import config
+
+
+def apply_field_scrambling(value: str) -> str:
+    if not value:
+        return value
+    ## TODO: use argparse to get min and max edit distance
```
[the remark this answers is not in the document]
Comment (m-goggins) [reply tests/load/scrambler/utils.py:11]: Yep, exactly. It's more important to have _any_ data than specific data.

[the remark this answers is not in the document]
Comment (bamader) [reply tests/load/scrambler/utils.py:11]: That's what I figured, makes sense to me!

File: tests/load/scrambler/json.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                if identifier["type"] == "SS":
+                    identifier["value"] = value
+                    break
+        elif field == "MRN":
+            for identifier in data.get("identifiers", []):
+                if identifier["type"] == "MR":
+                    identifier["value"] = value
+                    break
+    except KeyError:
+        pass
```
[the remark this answers is not in the document]
Comment (m-goggins) [reply tests/load/scrambler/json.py:171]: Yeah...I figured it was okay since it's about setting data which we don't have to be too picky about in this particular instance.

File: tests/load/scrambler/json.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+def prep_data(data: dict) -> dict:
+    """
+    Prepare the data for scrambling by ensuring all fields are present.
+    """
+    # Ensure all relevant fields are present in the data
+    for cluster in data["clusters"]:
+        for record in cluster["records"]:
+            for field in config.ALGORITHM_RELEVANT_COLUMNS:
+                if field not in record:
+                    record[field] = None
```
[the remark this answers is not in the document]
Comment (m-goggins) [reply tests/load/scrambler/json.py:49]: I hadn't thought of that. RL _should_ be able to handle either `None` or `''` for a given field.

[the remark this answers is not in the document]
Comment (m-goggins) [reply tests/load/scrambler/json.py:49]: oh nvm, this function was something I wanted to implement but didn't think we needed it in the end (at least not yet). Deleting for now.

File: tests/load/scripts/scramble_data.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    """
+    Main entry point for the script.
+    """
+    parser = argparse.ArgumentParser(description="Scramble test data")
+    parser.add_argument("--file", type=str, help="The file to load the test data from")
+    args = parser.parse_args()
+
+    if args.file.endswith(".json"):
+        with open(args.file, "r") as f:
+            data = json.load(f)
```
Comment (ericbuckley) [inline tests/load/scripts/scramble_data.py:18]: This is another place we'll have to keep an eye on.  I don't know if the plan was to load a file with 1.5 million records here, but if so, we might run into memory issues.

Comment (m-goggins) [reply tests/load/scripts/scramble_data.py:18]: Definitely!

File: tests/load/scrambler/json.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bamader) [reply tests/load/scrambler/json.py:155]: Oh completely agreed, it's absolutely fine here. Silently ignoring a problem makes so many things go away lol
